packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts

```diff
@@ -1403,25 +1596,17 @@ describeCompat("GC data store tombstone tests", "NoCompat", (getTestObjectProvid
 				// Requesting dataStore2 and dataStore3 should fail in container2.
 				const entryPoint2 = (await container2.getEntryPoint()) as ITestDataObject;
 				const containerRuntime2 = entryPoint2._context.containerRuntime as ContainerRuntime;
-				await assert.rejects(
-					async () => resolveHandleHelper(containerRuntime2, dataStore2._context.id),
-					(error: any) => {
-						const correctErrorType = error.code === 404;
-						const correctErrorMessage =
-							error.message === `DataStore was tombstoned: ${dataStore2._context.id}`;
-						return correctErrorType && correctErrorMessage;
-					},
-					`Should not be able to request dataStore2 which is tombstoned`,
+				assert.equal(
+					(containerRuntime2 as any).dataStores.contexts.get(dataStore2._context.id)
```

Comment:
Due to Auto-Recovery, validating an object is tombstoned by requesting it will change the course of these existing tests.  So instead, reach in to the DataStoreContext here.

Reply:
I should probably add a feature gate for the Auto-Recovery behavior, which could be used instead to keep these tests as-is.  Thoughts?

Reply:
I am not a fan of reaching out to the context directly here. It has a lot of assumptions baked in. Like the context will have a `tombstoned` property which can change.

I would say either disable the auto-recovery feature so we have coverage for both. Or update this to account for auto-recovery.

Reply:
Ok, that was my lean too but I wanted to get your input before making more changes.

> Or update this to account for auto-recovery.

I tried that first and it was a pain.

Reply:
Switched to disabling AutoRecovery